src/superclamp.coffee

```diff
@@ -32,9 +32,20 @@ class Superclamp
 
   @register: (nodeList) ->
     debug '.register', nodeList
+    # Subsequent nodes cannot be clamped in one iteration, as it would invalidate the bounding box computation
+    clampedNodes = []
+    deferredNodes = []
     for node in nodeList
-      @clamp(node)
+      previousSiblingIsBeingClamped = clampedNodes.indexOf(node.previousElementSibling) >= 0
+      previousSiblingWillBeClamped = deferredNodes.indexOf(node.previousElementSibling) >= 0
+      if previousSiblingIsBeingClamped || previousSiblingWillBeClamped
+        deferredNodes.push(node)
+      else
+        @clamp(node)
+        clampedNodes.push(node);
     drainQueue()
+    if deferredNodes.length > 0
+      @register(deferredNodes)
```

Comment:
I would suggest discarding the boolean variables since the check is now self-explanatory, and introducing a `previousNode` variable for extra clarity.

```suggestion
      previousNode = node.previousElementSibling
      if clampedNodes.has(previousNode) || deferredNodes.has(previousNode)
        deferredNodes.add(node)
      else
        @clamp(node)
        clampedNodes.add(node)
    drainQueue()
    if deferredNodes.length > 0
      @register(Array.from(deferredNodes))
```

(Using `Array.from` instead of `Set#entries` because of IE).

Reply:
`Set`s are awesome! Thanks for pointing out the horrible `O(n²)` thing, I should have seen that™

Some notes
* `Set#length` does not exist, `Set#size` does the job
* `Set` will drop IE < 11 support, if there was any before (I wouldn't care though)
* `Array#from` is undefined in IE11, neither is any other direct way to transform its values to an Array. We could either [use a workaround](https://stackoverflow.com/a/41261546), `Set#forEach` or just omit the transformation. Our code from the `register` function seems to handle Sets just as fine as Arrays.